sc-locked-asset-factory/src/locked_asset_token_merge.rs

```diff
@@ -0,0 +1,162 @@
+    + nft_deposit::NftDepositModule
+    + token_merge::TokenMergeModule
+{
+    #[endpoint(mergeLockedAssetTokens)]
+    fn merge_locked_asset_tokens(
+        &self,
+        #[var_args] opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<GenericTokenAmountPair<Self::BigUint>> {
+        let caller = self.blockchain().get_caller();
+        let deposit = self.nft_deposit(&caller).get();
+        require!(!deposit.is_empty(), "Empty deposit");
+
+        let (amount, attrs) = self.get_merged_locked_asset_token_amount_and_attributes(&deposit)?;
+        let locked_asset_token = self.locked_asset_token_id().get();
+
+        self.burn_deposit_tokens(&caller, &deposit);
+
+        self.nft_create_tokens(&locked_asset_token, &amount, &attrs);
+        self.increase_nonce();
+
+        let new_nonce = self.locked_asset_token_nonce().get();
+        self.send_nft_tokens(
+            &locked_asset_token,
+            new_nonce,
+            &amount,
+            &caller,
+            &opt_accept_funds_func,
+        )?;
+
+        Ok(GenericTokenAmountPair {
+            token_id: locked_asset_token,
+            token_nonce: new_nonce,
+            amount,
+        })
+    }
+
+    fn get_merged_locked_asset_token_amount_and_attributes(
+        &self,
+        deposit: &[GenericTokenAmountPair<Self::BigUint>],
+    ) -> SCResult<(Self::BigUint, LockedAssetTokenAttributes)> {
+        require!(!deposit.is_empty(), "Cannot merge with 0 tokens");
+
+        let mut tokens = Vec::new();
+        let mut sum_amount = 0u64.into();
+        let locked_asset_token_id = self.locked_asset_token_id().get();
+
+        for entry in deposit.iter() {
+            require!(entry.token_id == locked_asset_token_id, "Bad token id");
+
+            tokens.push(LockedToken {
+                token_amount: entry.clone(),
+                attributes: self.get_attributes(&entry.token_id, entry.token_nonce)?,
+            });
+            sum_amount += &entry.amount;
+        }
+
+        if tokens.len() == 1 {
+            return Ok((
+                tokens[0].token_amount.amount.clone(),
+                tokens[0].attributes.clone(),
+            ));
+        }
+
+        let attrs = LockedAssetTokenAttributes {
+            unlock_schedule: self.aggregated_unlock_schedule(&tokens)?,
+            is_merged: true,
+        };
+
+        Ok((sum_amount, attrs))
+    }
+
+    fn aggregated_unlock_schedule(
+        &self,
+        tokens: &[LockedToken<Self::BigUint>],
+    ) -> SCResult<UnlockSchedule> {
+        let mut unlock_epoch_amount = Vec::new();
+        tokens.iter().for_each(|locked_token| {
+            locked_token
+                .attributes
+                .unlock_schedule
+                .unlock_milestones
+                .iter()
+                .for_each(|milestone| {
+                    unlock_epoch_amount.push((
+                        milestone.unlock_epoch,
+                        self.rule_of_three(
```

Comment:
should we use rule_of_three_non_zero() here?

Reply:
Zero values should not be a problem as long as not all of them are zero